src/arch/Cargo.toml

```diff
@@ -5,11 +5,11 @@ authors = ["The Chromium OS Authors"]
 edition = "2018"
 
 [dependencies]
-kvm-bindings = { version = "0.3.0", features = ["fam-wrappers"] }
-kvm-ioctls = { version = "0.6.0" }
+kvm-bindings = { version = "=0.4.0", features = ["fam-wrappers"] }
```

Comment:
Why is this a fixed dependency now? Should you instead use `>=0.4.0` so that it is accurate in terms of minimum required version, but it is also not fixed and thus can be updated using `cargo update`?

Reply:
Good catch @andreeaflorescu!

It's weird that https://github.com/alindima/firecracker/blob/37af8903ae0aaf7241a36bc349d2f4330a328055/Cargo.lock#L855 shows that only `0.8.0` is used.

Maybe this is used only in `aarch64` and there are still some changes required there, this being one of them.

Reply:
AFAIK, we have to use version `0.4.0` in this toml file, since we `[patch]` this at the workspace level to use our internal fork: https://github.com/firecracker-microvm/firecracker/blob/37af8903ae0aaf7241a36bc349d2f4330a328055/Cargo.toml#L13
If somebody will perform a `cargo update` in the future to get version 5.0, they would bypass our patch without knowing it. Am I missing something?

Reply:
I think the cargo `[patch]` overrides everything. Unless that's not the case, keeping underlying dependencies flexible `>=` sounds like a good idea to me.

Reply:
If we leave `>=0.4.0` and upstream rust-vmm releases `0.5.0`, but we don't create a internal vmm-sys-util release in our fork, the `cargo update` will bypass the patch and give a warning:

```
warning: Patch `kvm-bindings v0.4.0 (https://github.com/firecracker-microvm/kvm-bindings?tag=v0.4.0-1#9b1dd38a)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
```

However, I think this is ok. This would force us to create an internal version corresponding to the upstream release. Either this, or modify our toml files.

Reply:
done